source/extensions/tracers/opentelemetry/resource_detectors/resource_provider.cc

```diff
@@ -14,23 +14,18 @@ namespace OpenTelemetry {
 namespace {
 bool isEmptyResource(const Resource& resource) { return resource.attributes_.empty(); }
 
-Resource createInitialResource(const std::string& service_name) {
+Resource createInitialResource(absl::string_view service_name) {
   Resource resource{};
 
   // Creates initial resource with the static service.name and telemetry.sdk.* attributes.
-  resource.attributes_[std::string(kServiceNameKey.data(), kServiceNameKey.size())] =
-      service_name.empty() ? std::string{kDefaultServiceName} : service_name;
-
-  resource
-      .attributes_[std::string(kTelemetrySdkLanguageKey.data(), kTelemetrySdkLanguageKey.size())] =
-      kDefaultTelemetrySdkLanguage;
+  if (!service_name.empty()) {
+    resource.attributes_[kServiceNameKey] = service_name;
```

Comment:
This would be a breaking change for the tracer? How about passing an optional string_view for now, so that we can keep tracer put unknown and metrics exporter put nothing?

Reply:
Ah, I didn't notice the default value in the header. Can you move that default to tracer itself? Header defaults are error-prone and easy to miss.

Reply:
done